### PR TITLE
Add org.bluesabre.MenuLibre

### PR DESCRIPTION
**Pull request by TheEvilSkeleton**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/bluesabre/menulibre/issues/65#issuecomment-955212964
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge). (Upstream maintainer does)
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

/cc @bluesabre

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by TheEvilSkeleton** [comment]
bot, build org.bluesabre.MenuLibre

File: org.bluesabre.MenuLibre.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  <launchable type="desktop-id">org.bluesabre.MenuLibre.desktop</launchable>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://raw.githubusercontent.com/bluesabre/menulibre/wiki-assets/readme/menulibre-github.png</image>
+    </screenshot>
+  </screenshots>
+  <releases>
+    <release version="2.2.3" date="2021-09-26"></release>
+  </releases>
+</component>
```
Comment (sonnyp) [inline org.bluesabre.MenuLibre.metainfo.xml:30]: upstream already has a metainfo

File: org.bluesabre.MenuLibre.yaml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+command: menulibre-wrapper
+rename-icon: menulibre
+rename-desktop-file: menulibre.desktop
+finish-args:
+  - '--filesystem=host:ro'
+  - '--socket=fallback-x11'
+  - '--socket=wayland'
+  - '--share=ipc'
+  - '--filesystem=xdg-data/applications:create'
+  - '--filesystem=~/.local/share/flatpak:ro'
```
Comment (sonnyp) [inline org.bluesabre.MenuLibre.yaml:14]:
```suggestion
  - '--filesystem=xdg-data/applications:create'
  - '--filesystem=xdg-data/flatpak:ro'
  - '--filesystem=/var/lib/flatpak:ro'
```

`--filesystem=xdg-data/flatpak:ro` is more standard compliant than `~/.local/share/flatpak`

`--filesystem=/var/lib/flatpak:ro` allows reading system flatpaks (host:ro doesn't cover it IIRC)

File: org.bluesabre.MenuLibre.yaml [hunk 3]
```diff
@@ -0,0 +1,107 @@
+app-id: org.bluesabre.MenuLibre
+runtime: org.gnome.Platform
+runtime-version: '40'
```
Comment (sonnyp) [inline org.bluesabre.MenuLibre.yaml:3]:
```suggestion
runtime-version: '41'
```

File: org.bluesabre.MenuLibre.metainfo.xml (hunk 1 above)
Comment (TheEvilSkeleton) [reply org.bluesabre.MenuLibre.metainfo.xml:30]: Huh yeah, I completely forgot about it. Thanks.

File: org.bluesabre.MenuLibre.yaml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        url: https://files.pythonhosted.org/packages/e1/b0/7276de53321c12981717490516b7e612364f2cb372ee8901bd4a66a000d7/psutil-5.8.0.tar.gz
+        sha256: 0c9ccb99ab76025f2f0bbecf341d4656e9c1351db8cc8a03ccd62e318ab4b5c6
+
+  - name: menulibre
+    buildsystem: simple
+    build-commands:
+      - 'python3 setup.py install --prefix=/app'
+    post-install:
+      - 'install -D menulibre-wrapper /app/bin/menulibre-wrapper'
+      - 'install -Dm644 org.bluesabre.MenuLibre.metainfo.xml /app/share/metainfo'
```
Comment (sonnyp) [inline org.bluesabre.MenuLibre.yaml:93]:
```suggestion
```

I just built it successfully without this

File: 0001-do-not-test-exec.patch [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# This is only required for flatpak
+# In the sandbox - the programs in the Exec and Try_Exec keys do not exist
+# it's not a problem when launching the app since we use flatpak-spawn
+# but when listing the apps - GLIb will filter them out
```
Comment (sonnyp) [inline 0001-do-not-test-exec.patch:4]:
```suggestion
# This is only required for flatpak
# In the sandbox - the programs in the Exec and Try_Exec keys do not exist
# when listing the apps - GLib will filter them out
```

makes more sense in the context of MenuLibre

**Comment by sonnyp** [comment]
bot, build org.bluesabre.MenuLibre

**Comment by TheEvilSkeleton** [comment]
Hmm... same issue.

**Comment by TheEvilSkeleton** [comment]
Looks like it's putting the metainfo in `/app/share/menulibre/metainfo`.
```
[📦 org.bluesabre.MenuLibre menulibre]$ find /app -name 'metainfo'
/app/share/menulibre/metainfo
```

**Comment by sonnyp** [comment]
Strange - it worked locally with

```
> flatpak-builder --version
flatpak-builder 1.2.0
> flatpak-builder --user --force-clean  --install-deps-from=flathub --install flatpak org.bluesabre.MenuLibre.yaml
```

**Comment by TheEvilSkeleton** [comment]
I just looked at the logs, and it looks so weird:

```
creating /app/share/metainfo
copying data/metainfo/menulibre.appdata.xml -> /app/share/metainfo/
creating /app/share/menulibre
creating /app/share/menulibre/media
copying data/media/menulibre.svg -> /app/share/menulibre/media
copying data/media/menulibre.png -> /app/share/menulibre/media
copying data/media/menulibre_32.svg -> /app/share/menulibre/media
copying data/media/menulibre_16.svg -> /app/share/menulibre/media
copying data/media/menulibre_48.svg -> /app/share/menulibre/media
copying data/media/menulibre_64.svg -> /app/share/menulibre/media
creating /app/share/menulibre/ui
copying data/ui/MenulibreWindow.ui -> /app/share/menulibre/ui
copying data/media/menulibre_24.svg -> /app/share/menulibre/media
creating /app/share/menulibre/metainfo
copying data/metainfo/menulibre.appdata.xml.in -> /app/share/menulibre/metainfo
```

**Comment by TheEvilSkeleton** [comment]
bot, build org.bluesabre.MenuLibre

**Comment by TheEvilSkeleton** [comment]
@sonnyp that was actually user error! I accidentally stated in the manifest to cleanup the metainfo directory. Now all we have to do is wait for https://github.com/bluesabre/menulibre/pull/90 to get merged.

**Comment by TheEvilSkeleton** [comment]
bot, build org.bluesabre.MenuLibre

**Comment by TheEvilSkeleton** [comment]
bot, build org.bluesabre.MenuLibre

**Comment by TheEvilSkeleton** [comment]
It works for me! Can anyone else test it?

**Comment by sonnyp** [comment]
Works for me :+1:

**Pull request closed without merging**